src/common/collection.c

```diff
@@ -535,26 +535,138 @@ GList *dt_collection_get_selected(const dt_collection_t *collection, int limit)
+  regex = g_regex_new("^\\s*(=|<|>|<=|>=|<>)?\\s*([0-9]+\\.?[0-9]*)\\s*$", 0, 0, NULL);
   g_regex_match_full(regex, input, -1, 0, 0, &match_info, NULL);
   match_count = g_match_info_get_match_count(match_info);
 
   if(match_count == 3)
   {
     *operator= g_match_info_fetch(match_info, 1);
-    *number = g_match_info_fetch(match_info, 2);
+    *number1 = g_match_info_fetch(match_info, 2);
+
+    if(*operator&& strcmp(*operator, "") == 0)
+    {
+      g_free(*operator);
+      *operator= NULL;
+    }
+  }
+
+  g_match_info_free(match_info);
+  g_regex_unref(regex);
+}
+void dt_collection_split_operator_datetime(const gchar *input, char **number1, char **number2, char **operator)
+{
+  GRegex *regex;
+  GMatchInfo *match_info;
+  int match_count;
+
+  *number1 = *number2 = *operator= NULL;
+
+  // we test the range expression first
+  // 4 elements : date1 time1(optional) date2 time2(optional)
+  regex = g_regex_new("^\\s*\\[\\s*(\\d{4}:\\d{2}:\\d{2})\\s*( "
+                      "\\d{2}:\\d{2}:\\d{2})?\\s*;\\s*(\\d{4}:\\d{2}:\\d{2})\\s*( "
+                      "\\d{2}:\\d{2}:\\d{2})?\\s*\\]\\s*$",
+                      0, 0, NULL);
+  g_regex_match_full(regex, input, -1, 0, 0, &match_info, NULL);
+  match_count = g_match_info_get_match_count(match_info);
+
+  if(match_count >= 4)
+  {
+    *number1 = g_match_info_fetch(match_info, 1);
+    gchar *number1_time = g_match_info_fetch(match_info, 2);
+    *number2 = g_match_info_fetch(match_info, 3);
+    *operator= g_strdup("[]");
+
+    if(!number1_time || strcmp(number1_time, "") == 0)
+      *number1 = dt_util_dstrcat(*number1, " 00:00:00");
+    else
+      *number1 = dt_util_dstrcat(*number1, "%s", number1_time);
+
+    if(match_count == 5)
+    {
+      gchar *number2_time = g_match_info_fetch(match_info, 4);
+      *number2 = dt_util_dstrcat(*number2, "%s", number2_time);
+      g_free(number2_time);
+    }
+    else
+      *number2 = dt_util_dstrcat(*number2, " 23:59:59");
+
+    if(number1_time) g_free(number1_time);
+    g_match_info_free(match_info);
+    g_regex_unref(regex);
+    return;
+  }
+
+  g_match_info_free(match_info);
+  g_regex_unref(regex);
+
+  // and we test the classic comparaison operators
+  // 2 elements : date time(optional)
+  regex = g_regex_new("^\\s*(=|<|>|<=|>=|<>)?\\s*(\\d{4}:\\d{2}:\\d{2})\\s*( \\d{2}:\\d{2}:\\d{2})?\\s*$", 0,
+                      0, NULL);
+  g_regex_match_full(regex, input, -1, 0, 0, &match_info, NULL);
+  match_count = g_match_info_get_match_count(match_info);
+
+  if(match_count >= 3)
+  {
+    *operator= g_match_info_fetch(match_info, 1);
+    *number1 = g_match_info_fetch(match_info, 2);
+    // we fill the time part if it's not set
+    if(match_count == 4)
+    {
+      gchar *nb_time = g_match_info_fetch(match_info, 3);
+      *number1 = dt_util_dstrcat(*number1, "%s", nb_time);
+      g_free(nb_time);
+    }
+    else if(*operator&& strcmp(*operator, ">") == 0)
+      *number1 = dt_util_dstrcat(*number1, " 23:59:59");
```

Comment:
That is interesting, as `man date` says that seconds can go to 60. But you are probably right and we can ignore it.